testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -779,6 +779,169 @@ public void testInsertOnlyMultipleWriters(boolean bucketed, HiveOrPresto inserte
+            onPresto().executeQuery(format("INSERT INTO %s VALUES (111, 'Katy', 57, 'CA'), (222, 'Joe', 72, 'WA')", tableName));
+            verifySelectForPrestoAndHive("SELECT * FROM " + tableName, "true", row(111, "Katy", 57, "CA"), row(222, "Joe", 72, "WA"));
+
+            onPresto().executeQuery(format("ALTER TABLE %s DROP COLUMN old_state", tableName));
+            log.info("This shows that neither Presto nor Hive see the old data after a column is dropped");
+            verifySelectForPrestoAndHive("SELECT * FROM " + tableName, "true", row(111, "Katy", 57), row(222, "Joe", 72));
+
+            onPresto().executeQuery(format("INSERT INTO %s VALUES (333, 'Kelly', 45)", tableName));
+            verifySelectForPrestoAndHive("SELECT * FROM " + tableName, "true", row(111, "Katy", 57), row(222, "Joe", 72), row(333, "Kelly", 45));
+
+            onPresto().executeQuery(format("ALTER TABLE %s ADD COLUMN new_state VARCHAR", tableName));
+            log.info("This shows that for ORC, Presto and Hive both see data inserted into a dropped column when a column of the same type but different name is added");
+            verifySelectForPrestoAndHive("SELECT * FROM " + tableName, "true", row(111, "Katy", 57, "CA"), row(222, "Joe", 72, "WA"), row(333, "Kelly", 45, null));
+        });
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL, dataProvider = "transactionModeProvider")
+    public void testOrcColumnTypeChange(boolean transactional)
+    {
+        ensureSchemaEvolutionSupported();
+        withTemporaryTable("test_orc_column_type_change", transactional, false, NONE, tableName -> {
+            onPresto().executeQuery(format("CREATE TABLE %s (id INT, old_name VARCHAR, age TINYINT, old_state VARCHAR) WITH (transactional = %s)", tableName, transactional));
+            onPresto().executeQuery(format("INSERT INTO %s VALUES (111, 'Katy', 57, 'CA'), (222, 'Joe', 72, 'WA')", tableName));
+            verifySelectForPrestoAndHive("SELECT * FROM " + tableName, "true", row(111, "Katy", 57, "CA"), row(222, "Joe", 72, "WA"));
+
+            onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN age age INT", tableName));
+            log.info("This shows that Hive see the old data after a column is widened");
+            assertThat(onHive().executeQuery("SELECT * FROM " + tableName))
+                    .containsOnly(row(111, "Katy", 57, "CA"), row(222, "Joe", 72, "WA"));
+            log.info("This shows that Trino gets an exception trying to widen the type");
+            assertThat(() -> onPresto().executeQuery("SELECT * FROM " + tableName))
+                    .failsWithMessageMatching(".*Malformed ORC file. Cannot read SQL type 'integer' from ORC stream '.*.age' of type BYTE with attributes.*");
+        });
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL)
+    public void testParquetColumnDropAdd()
```

Comment:
These tests have nothing to do with transactional tables, so please move them out from TestHiveTransactionalTable